src/core.c/Rakudo/Internals/PostcircumfixAdverbs.pm6

```diff
@@ -0,0 +1,509 @@
+      Array::Element::Access::not-exists-delete-p,
+      Array::Element::Access::not-exists-delete-not-p,
+      Array::Element::Access::delete,
+      Array::Element::Access::delete-kv,
+      Array::Element::Access::delete-not-kv,
+      Array::Element::Access::delete-p,
+      Array::Element::Access::delete-not-p,
+      Array::Element::Access::delete-k,
+      Array::Element::Access::delete-not-k,
+      Array::Element::Access::delete-v
+    );
+
+    method ACCESS-ELEMENT-DISPATCH-CLASS(int $index) {
+        nqp::atpos($access-element-dispatch,$index)
+    }
+
+    # Dispatch table for single element access in Any index
+    my constant $access-element-dispatch-any = nqp::list(
+      Array::Element::Access::none-any,
+      Array::Element::Access::kv-any,
+      Array::Element::Access::not-kv-any,
+      Array::Element::Access::p-any,
+      Array::Element::Access::not-p-any,
+      Array::Element::Access::k-any,
+      Array::Element::Access::not-k-any,
+      Array::Element::Access::v-any,
+      Array::Element::Access::exists-any,
+      Array::Element::Access::exists-kv-any,
+      Array::Element::Access::exists-not-kv-any,
+      Array::Element::Access::exists-p-any,
+      Array::Element::Access::exists-not-p-any,
+      Array::Element::Access::exists-delete-any,
+      Array::Element::Access::exists-delete-kv-any,
+      Array::Element::Access::exists-delete-not-kv-any,
+      Array::Element::Access::exists-delete-p-any,
+      Array::Element::Access::exists-delete-not-p-any,
+      Array::Element::Access::not-exists-any,
+      Array::Element::Access::not-exists-kv-any,
+      Array::Element::Access::not-exists-not-kv-any,
+      Array::Element::Access::not-exists-p-any,
+      Array::Element::Access::not-exists-not-p-any,
+      Array::Element::Access::not-exists-delete-any,
+      Array::Element::Access::not-exists-delete-kv-any,
+      Array::Element::Access::not-exists-delete-not-kv-any,
+      Array::Element::Access::not-exists-delete-p-any,
+      Array::Element::Access::not-exists-delete-not-p-any,
+      Array::Element::Access::delete-any,
+      Array::Element::Access::delete-kv-any,
+      Array::Element::Access::delete-not-kv-any,
+      Array::Element::Access::delete-p-any,
+      Array::Element::Access::delete-not-p-any,
+      Array::Element::Access::delete-k-any,
+      Array::Element::Access::delete-not-k-any,
+      Array::Element::Access::delete-v-any
+    );
+
+    method ACCESS-ELEMENT-ANY-DISPATCH-CLASS(int $index) {
+        nqp::atpos($access-element-dispatch-any,$index)
+    }
+
+    # Constants to allow mapping of valid slice adverb combinations to
+    # a value that can be used in lookup table to get a dispatch table
+    # lookup value.
+    my constant SLICE_NO_ADVERBS = 0x0000;
+    my constant SLICE_DELETE     = 0x0001;  # :delete
+    my constant SLICE_EXISTS     = 0x0002;  # :exists
+    my constant SLICE_NOT_EXISTS = 0x0004;  # :!exists
+    my constant SLICE_KV         = 0x0008;  # :k
```

Comment:
Very minor remark: The comment should be ```:kv``` here. (I just stumbled upon it.)

Reply:
Addressed with 5a6f6cf907be731961f50a56